docs/CONFIG.md

```diff
@@ -0,0 +1,165 @@
+# Rollup BSN Contract – Configuration Specification
+
+## Overview
+This document defines the parameters accepted during instantiation
+of the Rollup BSN contract. Each parameter is described in terms of
+its role within the system, expected format, and configuration considerations.
+These values are set once at deployment and directly influence the contract’s behavior and validation rules.
+
+To learn how to instantiate the contract, see the [deployment guide](./deployment-integration.md)
+
+### Instantiation Parameters
+
+The following parameters must be provided when deploying the Rollup BSN
+contract:
+
+- `admin`: Babylon Genesis address of the contract administrator
+- `bsn_id`: Unique identifier for the BSN rollup
+- `allowed_finality_providers`: List of BTC public keys authorized to submit
+  finality signatures and randomness
+- `bsn_activation_height`: Rollup block height at which finality becomes
+  active
+- `finality_signature_interval`: Interval (in rollup blocks) that determines
+  which blocks are eligible for finality signatures
+- `min_pub_rand`: Minimum number of public randomness values submitted per
+  commit
+- `rate_limiting_interval`: Block interval used to enforce rate limiting on
+  finality provider messages
+- `max_msgs_per_interval`: Maximum number of messages a finality provider can
+  submit within a rate limiting interval
+
+The section below defines each parameter in detail. It includes the expected
+format, its purpose within the system, and configuration considerations.
+
+---
+
+#### `admin`
+
+The Babylon address with administrative control over the contract.
+
+This address is authorized to perform privileged actions such as modifying the
```

Comment:
Shall we note that this address can also be the governance of Babylon Genesis but this might lead to slow modifications and reaction times?